diff --git a/tests/columns/test_json.py b/tests/columns/test_json.py
index 03ea0ff..33a85db 100644
--- a/tests/columns/test_json.py
+++ b/tests/columns/test_json.py
@@ -13,11 +13,11 @@ class JSONTestCase(BaseTestCase):
         return {'allow_experimental_object_type': 1}
 
     def test_simple(self):
-        rv = self.client.execute("SELECT '{\"bb\": {\"cc\": [255, 1]}}'::JSON")
+        rv = self.client.execute("SELECT '{\"bb\": {\"cc\": [255, 1]}}'::Object('json')")
         self.assertEqual(rv, [({'bb': {'cc': [255, 1]}},)])
 
     def test_from_table(self):
-        with self.create_table('a JSON'):
+        with self.create_table("a Object('json')"):
             data = [
                 ({},),
                 ({'key1': 1}, ),
@@ -46,7 +46,7 @@ class JSONTestCase(BaseTestCase):
             self.assertEqual(inserted, data_with_all_keys)
 
     def test_insert_json_strings(self):
-        with self.create_table('a JSON'):
+        with self.create_table("a Object('json')"):
             data = [
                 (json.dumps({'i-am': 'dumped json'}),),
             ]
@@ -67,7 +67,7 @@ class JSONTestCase(BaseTestCase):
         settings = {'namedtuple_as_json': False}
         query = 'SELECT * FROM test'
 
-        with self.create_table('a JSON'):
+        with self.create_table("a Object('json')"):
             data = [
                 ({'key': 'value'}, ),
             ]
