diff --git a/tests/numpy/test_generic.py b/tests/numpy/test_generic.py
--- a/tests/numpy/test_generic.py
+++ b/tests/numpy/test_generic.py
@@ -57,7 +57,7 @@ class NumpyProgressTestCase(NumpyBaseTestCase):
         progress = self.client.execute_with_progress('SELECT 2')
         self.assertEqual(
             list(progress),
-            [(1, 0), (1, 0)] if self.server_version > (20,) else [(1, 0)]
+            [(1, 1), (1, 1)] if self.server_version > (20,) else [(1, 0)]
         )
         self.assertEqual(progress.get_result(), [(2,)])
         self.assertTrue(self.client.connection.connected)
