diff --git a/tests/testcase.py b/tests/testcase.py
index f44c902..3a08aea 100644
--- a/tests/testcase.py
+++ b/tests/testcase.py
@@ -80,7 +80,7 @@ class BaseTestCase(BaseAbstractTestCase, TestCase):
         version = system_native_session.execute(
             text('SELECT version()')
         ).fetchall()
-        cls.server_version = tuple(int(x) for x in version[0][0].split('.'))
+        cls.server_version = tuple(int(x) for x in version[0][0].split('.')[0:-1])
 
         super().setUpClass()
 
@@ -110,7 +110,7 @@ class BaseAsynchTestCase(BaseTestCase):
         version = (
             await system_asynch_session.execute(text('SELECT version()'))
         ).fetchall()
-        cls.server_version = tuple(int(x) for x in version[0][0].split('.'))
+        cls.server_version = tuple(int(x) for x in version[0][0].split('.')[0:-1])
 
         super(BaseTestCase, cls).setUpClass()
 
